pysidt/sidt.py

```diff
@@ -1102,7 +1104,20 @@ def estimate_uncertainty(self,rel_node_dof_tolerance=1e-5):
                 node.rule.uncertainty = node.parent.rule.uncertainty
             elif node.rule.num_data == 0:
                 node.rule.uncertainty = 0.0 #if n=0 the LASSO should drive node.rule.value to zero so there should be approximately no variance contribution 
-
+                
+        if self.validation_set and len(node_uncertainties) > 0:
+            val_predictions_uncertainties = [self.evaluate(d.mol, estimate_uncertainty=True) for d in self.validation_set]
```

Comment:
We probably only want to run this once perhaps at the end of generation so either we put this is in a separate function "scale_uncertainties" or add a flag to this function to not do this so we can not scale them everytime we call this during tree generation except one final time at the end.